src/main/resources/project/project.xml

```diff
@@ -119,6 +119,12 @@
                         <expandable>1</expandable>
                         <value/>
                     </property>
+					<property>
+                        <propertyName>PuppetAgentSetConfig</propertyName>
+                        <description>PuppetAgentSetConfig.</description>
+                        <expandable>1</expandable>
+                        <value/>
```

Comment:
I had merged this PR earlier but can we still address my earlier comment:

I think we should also call this procedure in a step in 'ConfigureAgent' so that we can use this proc to set the puppet master, environment and other details.